src/ImpactEvaluator.sol

```diff
@@ -88,45 +87,50 @@ contract ImpactEvaluator is AccessControl {
         Round storage round = openRounds[roundIndex];
         require(round.exists, "Open round does not exist");
 
-        for (uint i = 0; i < addresses.length; i++) {
-            round.addresses.push(addresses[i]);
-            round.scores.push(scores[i]);
-        }
+        uint sumOfScores = validateScores(scores, round.totalScores);
+        reward(addresses, scores);
+        round.totalScores += sumOfScores;
 
-        if (allScoresSubmitted(round.scores)) {
-            reward(round.addresses, round.scores);
-            cleanUpRound(round, roundIndex);
+        if (round.totalScores == MAX_SCORE) {
+            delete openRounds[roundIndex];
         }
     }
 
-    function cleanUpRound(Round storage round, uint roundIndex) private {
-        round.end = 0;
-        delete round.addresses;
-        delete round.scores;
-        round.exists = false;
-        delete openRounds[roundIndex];
-    }
-
-    function allScoresSubmitted(
-        uint64[] memory scores
-    ) private pure returns (bool) {
-        uint totalScore = 0;
+    function validateScores(
+        uint64[] memory scores,
+        uint scoresAlreadySubmitted
+    ) private pure returns (uint) {
+        uint64 sum = 0;
         for (uint i = 0; i < scores.length; i++) {
-            totalScore += scores[i];
+            sum += scores[i];
         }
-        return totalScore >= MAX_SCORE;
+        require(sum <= MAX_SCORE, "Sum of scores too big");
```

Comment:
I thought that we always wanted the scores to add up to some fixed number so we can distribute the funds relative to the scores. (For example all the scores should add up to 100)

Shouldn't we be checking that here?

Reply:
Say we call `setScores()` four times, the first the times are ok and the fourth time makes the totalScore overflow. Don't you think we should revert here? Or should we ignore the too high scores and perform a partial payout?